src/compiler/target/golang/mod.rs

```diff
@@ -0,0 +1,268 @@
+use super::*;
+
+pub struct Go {}
+
+impl Go {
+  /// Generate golang code using tokens from parser
+  pub fn generate(lb: &mut LangBuilder, t: AnilizedTokens) -> Result<(), LocationError> {
+
+    // TODO: Replace when file importing is implemented
+    lb.code("package main");
+
+    let mut code = Self {};
+
+    // define functions
+    for (_, func) in t.functions {
+      code.function(func, lb);
+    }
+
+    // define types
+    for (_, type_) in t.types {
+      code.code_type(type_, lb);
+    }
+
+    // define structs
+    for (_, structure) in t.structs {
+      code.structure(structure, lb);
+    }
+    
+    // define globals
+    for (_, glob) in t.vars {
+      code.global_var(glob, lb);
+    }
+
+    Ok(())
+  }
+  /// Get the type as a string
+  pub fn get_type(&mut self, type_: Option<tokenize::types::Type>) -> String {
+    match type_ {
+      Some(res) => {
+        match res.type_ {
+          tokenize::types::TypeType::Char => "string".to_string(),
```

Comment:
Golang also has a char type, it works mostly the same as the one from rust

Reply:
Would this be a `byte`? I can't find a type `char` anywhere

Reply:
Wow i just got completely confused,  
I was like what all this time i have been using chars in go but they are just bytes????  

But in GoLang the char type is named `rune`.  
In Go a rune can be defined using: 'a'.  

Some side notes a rune can be compared with another rune and a byte *(`"a"[0] == 'a'` = true)*.  
A rune can be re-mapped to byte if the char is a ASCII letter.

Reply:
Ah, ok. I’ll update it with the rune later ;-)